### PR TITLE
devmode: make HOSTNAME and PRIVATE_IP configurable

### PR DESCRIPTION
**Pull request by hunchback**
To test that there is not degradation run:

```
export PREPARE_BOX=true
vagrant up
```

Next run:

```
export HOSTNAME=dev2
export PRIVATE_IP=192.168.10.11
export PREPARE_BOX=true
vagrant up
```

### CODE REVIEW

**Comment by hunchback** [comment]
run skydive-ppc64le-tests

**Comment by hunchback** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-create-vagrant-boxes

**Comment by nplanel** [comment]
run skydive-ppc64le-tests

**Comment by hunchback** [comment]
@lebauce - `skydive-create-vagrant-boxes` is failing due to:

```
failed: [dev] (item=openvswitch) => {"changed": false, "item": "openvswitch", "msg": "Unable to start service openvswitch: A dependency job for openvswitch.service failed. See 'journalctl -xe' for details.\n"}
```

Please check

**Comment by lebauce** [comment]
@hunchback This is not a problem specific to CI. If you build to image on your machine, you'll face the same problem

File: contrib/dev/Vagrantfile [hunk 1]
```diff
@@ -135,8 +139,8 @@ def dev_synced_folders(override, type)
 end
 
 Vagrant.configure(2) do |config|
-  config.vm.define "dev" do |dev|
-    dev.vm.hostname = "dev"
+  config.vm.define HOSTNAME do |dev|
```
Comment (lebauce) [inline contrib/dev/Vagrantfile]: The name of the VM should not change and stay "dev"

**Comment by lebauce** [comment]
@hunchback The issue
```
failed: [dev] (item=openvswitch) => {"changed": false, "item": "openvswitch", "msg": "Unable to start service openvswitch: A dependency job for openvswitch.service failed. See 'journalctl -xe' for details.\n"}
```
was fixed by https://github.com/skydive-project/skydive/pull/1266

File: contrib/dev/Vagrantfile (hunk 1 above)
Comment (hunchback) [reply contrib/dev/Vagrantfile]: left hostname "dev" but I allow to change the VM Name

File: contrib/dev/Vagrantfile [hunk 2]
```diff
@@ -1,8 +1,11 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
+HOSTNAME = ENV.fetch("HOSTNAME", "dev")
+MEMORY = ENV.fetch("MEMORY", "3072")
```
[the remark this answers is not in the document]
Comment (hunchback) [reply contrib/dev/Vagrantfile]: applied

File: contrib/dev/Vagrantfile [hunk 3]
```diff
@@ -11,7 +14,7 @@ end
 DEVMODE_ROOT = "../.."
 GOPATH = "/home/vagrant"
 
-SKYDIVE_SYNC_FOLDER = ENV.fetch("SKYDIVE_SYNC_FOLDER", "true") == "true" ? true : false
+SKYDIVE_SYNC_FOLDER = ENV.fetch("SKYDIVE_SYNC_FOLDER", (Vagrant::Util::Platform.windows? ? "false" : "true")) == "true" ? true : false
```
[the remark this answers is not in the document]
Comment (hunchback) [reply contrib/dev/Vagrantfile]: set default to "false" for all

File: contrib/dev/Vagrantfile [hunk 4]
```diff
@@ -1,8 +1,11 @@
 # -*- mode: ruby -*-
 # vi: set ft=ruby :
 
+HOSTNAME = ENV.fetch("HOSTNAME", "dev")
```
[the remark this answers is not in the document]
Comment (hunchback) [reply contrib/dev/Vagrantfile]: I replaced to VMNAME - so that one can label the VM - for example "dev-experimental" or "dev2"

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests

**Comment by lebauce** [comment]
run skydive-create-vagrant-boxes

**Comment by hunchback** [comment]
run skydive-cdd-overview-tests

**Comment by hunchback** [comment]
run skydive-create-vagrant-boxes

**Comment by hunchback** [comment]
run skydive-cdd-overview-tests

**Pull request merged**